packages/react-components/react-menu/src/components/MenuTrigger/MenuTrigger.types.ts

```diff
@@ -12,16 +12,7 @@ export type MenuTriggerProps = {
 /**
  * Props that are passed to the child of the MenuTrigger when cloned to ensure correct behaviour for the Menu
  */
-export type MenuTriggerChildProps = Required<
-  Pick<
-    React.HTMLAttributes<HTMLElement>,
-    'onClick' | 'onMouseEnter' | 'onMouseLeave' | 'onMouseMove' | 'onContextMenu' | 'onKeyDown' | 'aria-haspopup' | 'id'
-  >
-> & {
-  ref?: React.Ref<never>;
-  // aria-expanded should only be present when the menu when it is not a submenu
-  'aria-expanded': boolean | undefined;
-};
+export type MenuTriggerChildProps = React.HTMLAttributes<HTMLElement>;
```

Comment:
Generalizes type for `TriggerChildProps` as we're using `useARIAButton` internally which means properties types are not 100% certain to be there, it might depend on the type of the `children` being passed to the `Trigger`

Reply:
The annoying part of generalizing the types is that the render function API is not not very clear what props it is being called with

```tsx
<MenuTrigger>
  {(triggerProps) => {....}}
</MenuTrigger>
```